### DISCUSSION TITLE
Backup, downgrade and full reset ?

### DISCUSSION THREAD

**Discussion by chrbar11** [roleoroleo/yi-hack-Allwinner-v2#645] [Q&A] [answered]
Thanks a lot Roleo for your great job with Yi-Hack.

I did some test about Yi firmware backup, downgrade and full reset.
I think I've found some answers to my questions, but I would like to double check here... that could help someone too :)

**1/ Backup of the native Yi Firmware (without hack)**

In [Installation page](https://github.com/roleoroleo/yi-hack-Allwinner-v2#installation), it's recommanded [to make a backup copy](https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Dump-your-backup-firmware-(SD-card)) before run the hack.

Yi-Hack releases and "Install Procedure" include "Factory" folder.
Does it meam that a backup of the native Yi Firmware (without hack), will be done as first step, when the first hack is applied?

**2/ Downgrade Yi Firmware**

In case of my Yi Camera came with the built-in 9.0.26* Yi Firmware version, and I did a backup of this Yi Firmware by following [Dump your backup firmware (SD card)](https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Dump-your-backup-firmware-(SD-card)) process  (or via "Factory" folder from the first hack).

If I did an Yi Firmware update to 11.1.26* (from Yi App), can I downgrade Yi Firmware to 9.0.26* by following [Unbrick the cam](https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Unbrick-the-cam) soft way process?

**3/ Downgrade Yi-Hack Firmware**

If I hack my Yi Camera with Yi-Hack 0.2.6 version.
Can I downgrade Yi-Hack to 0.2.0 version?

To do this Yi-Hack Firmware downgrade, can I simplely run a new Yi-Hack 0.2.0 (as first hack) on the Yi-Hack 0.2.6 camera?
Or do I've to follow [Unbrick the cam](https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Unbrick-the-cam) soft way process firstly, and run a new Yi-Hack 0.2.0 after?

**4/ Camera full reset**

To reset a camera like straight out of the box, I've to follow [Unbrick the cam](https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Unbrick-the-cam) soft way process 

But do I have also to use the reset hole/pin on camera (as discribed on the below video)  to do a full camera reset?
https://help.yitechnology.com/hc/en-us/articles/360045037674-How-to-reset-your-YI-Home-camera-

Thanks a lot for your help :)
Chris

**Comment by roleoroleo** [comment, marked as answer]
> Thanks a lot Roleo for your great job with Yi-Hack.
> 
> I did some test about Yi firmware backup, downgrade and full reset. I think I've found some answers to my questions, but I would like to double check here... that could help someone too :)
> 
> **1/ Backup of the native Yi Firmware (without hack)**
> 
> In [Installation page](https://github.com/roleoroleo/yi-hack-Allwinner-v2#installation), it's recommanded [to make a backup copy](https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Dump-your-backup-firmware-(SD-card)) before run the hack.
> 
> Yi-Hack releases and "Install Procedure" include "Factory" folder. Does it meam that a backup of the native Yi Firmware (without hack), will be done as first step, when the first hack is applied?

Yes, a backup is automatically created when the hack starts. In the "backup" folder of the sd card.
Check the code: https://github.com/roleoroleo/yi-hack-Allwinner-v2/blob/master/sdhack/h30ga/Factory/config.sh
Or you can create a backup following the wiki page: https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Dump-your-backup-firmware-(SD-card)

> **2/ Downgrade Yi Firmware**
> 
> In case of my Yi Camera came with the built-in 9.0.26* Yi Firmware version, and I did a backup of this Yi Firmware by following [Dump your backup firmware (SD card)](https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Dump-your-backup-firmware-(SD-card)) process (or via "Factory" folder from the first hack).
> 
> If I did an Yi Firmware update to 11.1.26* (from Yi App), can I downgrade Yi Firmware to 9.0.26* by following [Unbrick the cam](https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Unbrick-the-cam) soft way process?

Yes.
The number of partitions to restore depends on the difference between the firmware installed and the firmware backup.
A full restore requires to use backup, home and rootfs.

> **3/ Downgrade Yi-Hack Firmware**
> 
> If I hack my Yi Camera with Yi-Hack 0.2.6 version. Can I downgrade Yi-Hack to 0.2.0 version?
> 
> To do this Yi-Hack Firmware downgrade, can I simplely run a new Yi-Hack 0.2.0 (as first hack) on the Yi-Hack 0.2.6 camera? Or do I've to follow [Unbrick the cam](https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Unbrick-the-cam) soft way process firstly, and run a new Yi-Hack 0.2.0 after?

Usually it is enough to erase the sd card and run the new hack.
If this doesn't work it's better to restore the cam to the original state following the unbrick procedure.

> **4/ Camera full reset**
> 
> To reset a camera like straight out of the box, I've to follow [Unbrick the cam](https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Unbrick-the-cam) soft way process
> 
> But do I have also to use the reset hole/pin on camera (as discribed on the below video) to do a full camera reset? https://help.yitechnology.com/hc/en-us/articles/360045037674-How-to-reset-your-YI-Home-camera-
> 
> Thanks a lot for your help :) Chris

The unbrick procedure restore the firmware to the original status (factory firmware) but doesn't change the configuration of the cam.
If you want to clean the configuration and restart the pairing process you need to reset the cam with the hole/pin.
